src/compiler/types.rs

```diff
@@ -0,0 +1,185 @@
+use std::rc::Rc;
+
+pub struct Path(String);
+
+impl Path {
+    pub fn new(string: String) -> Self {
+        Self(string)
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum SyntaxKind {
+    Unknown,
+    EndOfFileToken,
+    SemicolonToken,
+    AsteriskToken,
+    EmptyStatement,
+    SourceFile,
+}
+
+#[derive(Debug)]
+pub struct BaseNode {
+    pub kind: SyntaxKind,
+}
+
+pub trait NodeInterface {
+    fn kind(&self) -> SyntaxKind;
+}
+
+#[derive(Debug)]
+pub enum Node {
+    Statement(Statement),
+}
+
+impl NodeInterface for Node {
+    fn kind(&self) -> SyntaxKind {
+        match self {
+            Node::Statement(statement) => statement.kind(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NodeArray {
+    _nodes: Vec<Node>,
+}
+
+impl NodeArray {
+    pub fn new(nodes: Vec<Node>) -> Self {
+        NodeArray { _nodes: nodes }
+    }
+}
```

Comment:
If the only point of this is to be consistent with the TS naming, you could use a type alias to `Vec<Node>` rather than wrapping it in a struct, which would get rid of all the converting stuff.

Reply:
There will be other fields added to `NodeArray`